### PR TITLE
1373 update hets instances regularly

### LINKED ISSUE #1373: Update HetsInstance regularly
Create a Sidekiq job to check the HetsInstances and update them if needed.

### PR DESCRIPTION
**Pull request by 0robustus1**
Shall fix #1373.
Please pay special attention to validation of the new setting, as i am currently not sure if that is the right way to add to the settings validations.

### CODE REVIEW

File: lib/hets_instance_worker.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+require 'sidekiq/worker'
+
+class HetsInstanceWorker < BaseWorker
+  def self.schedule_update(hets_instance_id)
+    update_span = Settings.hets.time_between_updates.hours
+    perform_in(update_span, hets_instance_id)
```
Comment (eugenk) [inline lib/hets_instance_worker.rb]: I think we should specify a queue to ensure that it doesn't get into the unnamed queue.

File: db/seeds/001-hets.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
-%w(localhost:8000).each do |uri|
-  HetsInstance.create(name: uri, uri: "http://#{uri}")
+Sidekiq::Testing.disable! do
```
Comment (eugenk) [inline db/seeds/001-hets.rb]: Why don't you want this to be executed inline?

Comment (eugenk) [inline db/seeds/001-hets.rb]: Nevermind... You commit message says it.

File: spec/hets_helper.rb [hunk 3]
```diff
@@ -99,14 +99,18 @@ def stub_hets_for(ontology_fixture,
...
-    FactoryGirl.create(:local_hets_instance)
-    hets_instance = HetsInstance.choose!
-  end
-ensure
+  hets_instance =
+    begin
+      HetsInstance.choose!
+    rescue HetsInstance::NoRegisteredHetsInstanceError => e
+      create_local_hets_instance
+      hets_instance = HetsInstance.choose!
```
Comment (eugenk) [inline spec/hets_helper.rb]: `hets_instance = ` in the last line is redundant

File: spec/hets_helper.rb [hunk 4]
```diff
@@ -99,14 +99,18 @@ def stub_hets_for(ontology_fixture,
...
-rescue HetsInstance::NoRegisteredHetsInstanceError => e
-  if hets_instance.nil?
-    FactoryGirl.create(:local_hets_instance)
-    hets_instance = HetsInstance.choose!
-  end
-ensure
+  hets_instance =
+    begin
+      HetsInstance.choose!
+    rescue HetsInstance::NoRegisteredHetsInstanceError => e
```
Comment (eugenk) [inline spec/hets_helper.rb]: `=> e` is redundant

**Comment by 0robustus1** [comment]
Your comments on code have been integrated (through commit-amends and rebase).
2165361 sets the queue and 2767a98 switches from hours to minutes for the settings (i left the validation as is).

**Comment by eugenk** [comment]
:+1:

**Pull request merged**